src/Hit/Issue.hs

```diff
@@ -164,3 +162,24 @@ parseOwnerRepo url =
 
     stripGitSuffix :: Text -> Maybe Text
     stripGitSuffix x = whenNothing (T.stripSuffix ".git" x) (Just x)
+
+data AssignSuccess = AlreadyAssigned | NewlyAssigned
+
+assignToIssue :: Issue -> IO ()
+assignToIssue issue = withOwnerRepo assignAction >>= \case
+    Left err              -> errorMessage (T.pack $ show err) >> exitFailure
+    Right AlreadyAssigned -> successMessage "Already assigned to issue"
+    Right NewlyAssigned   -> successMessage "Assigned to issue"
+  where
+    assignAction Nothing _ _ = pure $ Left $ UserError "GITHUB_TOKEN failed"
```

Comment:
Let's add a type signature to this function. It looks quite complicated and it's not obvious what's going on here without types